pos_accounting/src/pool/delta/delta_amount_collection.rs

```diff
@@ -62,52 +67,19 @@ impl<K: Ord> DeltaAmountCollection<K> {
         }
     }
 
-    pub fn add_signed(&mut self, id: K, to_add: SignedAmount) -> Result<(), Error> {
-        match self.data.entry(id) {
-            std::collections::btree_map::Entry::Vacant(e) => {
-                e.insert(to_add);
-            }
-            std::collections::btree_map::Entry::Occupied(mut e) => {
-                let current_amount = e.get();
-                let new_amount = (*current_amount + to_add).ok_or(Error::AdditionError)?;
-                if new_amount != SignedAmount::ZERO {
-                    let _ = e.insert(new_amount);
-                } else {
-                    let _ = e.remove();
-                }
-            }
-        }
-        Ok(())
-    }
-
     pub fn add_unsigned(&mut self, id: K, to_add: Amount) -> Result<(), Error> {
         let signed_amount_to_add =
             to_add.into_signed().ok_or(Error::ArithmeticErrorToSignedFailed)?;
-        self.add_signed(id, signed_amount_to_add)
-    }
-
-    pub fn sub_signed(&mut self, id: K, to_sub: SignedAmount) -> Result<(), Error> {
-        match self.data.entry(id) {
-            std::collections::btree_map::Entry::Vacant(e) => {
-                e.insert(to_sub);
-            }
-            std::collections::btree_map::Entry::Occupied(mut e) => {
-                let current_amount = e.get();
-                let new_amount = (*current_amount - to_sub).ok_or(Error::SubError)?;
-                if new_amount != SignedAmount::ZERO {
-                    let _ = e.insert(new_amount);
-                } else {
-                    let _ = e.remove();
-                }
-            }
-        }
-        Ok(())
+        self.merge_delta_amount_element(id, signed_amount_to_add)
     }
 
     pub fn sub_unsigned(&mut self, id: K, to_sub: Amount) -> Result<(), Error> {
```

Comment:
I'm confused. You mentioned something about having reduntant methods for adding and subtracting and I agreed to removing some of them, but now we have no methods to add signed amount, which means a user of the library will have to manually go for adding a negative number to subtracting a positive number, which is really not a negligible amount of work and heavily error prone. I was OK with keeping an add_signed and removing the sub_signed because flipping the sign is not that hard (it's not that simple either since there's error handling involved), but now the user has to match and choose the method to call and attempt to safely convert from a negative SignedAmount to a positive Amount without losing the sign information somehow. Am I getting this right?

Reply:
Not exactly. We discussed `sub_signed` but while writing tests I noticed that `merge_delta_amount_element` is an exact equivalent of `addSigned`. So I used it in code and removed redundant *_signed functions.  And now we have `merge_delta_amount_element` for signed operations and add/sub_unsigned (3 instead of 5 functions)

Reply:
btw there is `merge_delta_data_element` function which is a good match for `merge_delta_amount_element`